### ISSUE TITLE
TSServer Should Mark Const Arrow Function with 'Function' Kind

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#14830]
From https://github.com/Microsoft/vscode/issues/19401

**TypeScript Version:**  2.2.1
**Code**

```ts
export var abc = () => x;

abc|
```

Run `completions` at the `|`

**Expected behavior:**
In `completions`, `abc` is marked as a function

**Actual behavior:**
`abc` has a `const` kind:

```
[Trace - 4:11:51 PM] Sending request: completions (46). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/sand/c.js",
    "line": 3,
    "offset": 4
}
[Trace - 4:11:51 PM] Response received: completions (46). Request took 2 ms. Success: true 
Result: [
    {
        "name": "abc",
        "kind": "const",
        "kindModifiers": "export",
        "sortText": "0"
    }
```

This differs from `navtree` which marks `abc` as a `function`:

```
Trace - 4:12:25 PM] Sending request: navtree (57). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/sand/c.js"
}
[Trace - 4:12:25 PM] Response received: navtree (57). Request took 1 ms. Success: true 
Result: {
    "text": "\"c\"",
    "kind": "module",
    "kindModifiers": "",
    "spans": [
        {
            "start": {
                "line": 1,
                "offset": 1
            },
            "end": {
                "line": 3,
                "offset": 4
            }
        }
    ],
    "childItems": [
        {
            "text": "abc",
            "kind": "function",
            "kindModifiers": "export",
            "spans": [
                {
                    "start": {
                        "line": 1,
                        "offset": 18
                    },
                    "end": {
                        "line": 1,
                        "offset": 25
                    }
                }
            ]
        }
    ]
}
```

**Comment by mhegazy** [comment]
I think we are tying to be too smart with the nav-tree. `abc` is a `const`, and not a `function`. the fix here i would say is to undo the nav-tree categorizing variables as functions.

**Comment by unional** [comment]
The same goes for `const` + function expression: `const abc = function () { ... }`

For tooling usability, able to detect them as functions is valuable.

**Comment by andy-ms** [comment]
We don't seem to return `function` as the kind for `export var abc = () => x;` any more. However, for `export var abc = function f() {}`, we will return an item for the function but *not* for the var. Don't think there's any work to do here.

**Comment by andy-ms** [comment]
Related issue that was closed: #25126
